docs/howto.rst

```diff
@@ -49,15 +49,19 @@ parameters that may already exist.
 The ``params`` argument may be either a ``dict`` or a ``list`` of
 ``(key, value)`` tuples.
 
-If it is a ``dict`` then the values in the dict may either be a ``str`` value
-or a ``list`` of ``str`` values.
+If it is a ``dict`` then the values in the dict may either be scalar values or a ``list`` or ``tuple`` thereof.
+Scalar values means ``str``, ``bytes``, or anything else — even ``None`` — which will be coerced to ``str``.
+Strings are UTF-8 encoded.
 
 .. literalinclude:: examples/query_params.py
     :linenos:
     :lines: 7-37
 
 Full example: :download:`query_params.py <examples/query_params.py>`
 
+If you prefer a strictly-typed API, try :class:`hyperlink.DecodedURL`.
```

Comment:
Should this be hyperlink.EncodedURL ?

I think that examples/query_params.py or inline here, add an example of using hyperlink.EncodedURL together with treq.

```python
from hyperlink import EncodedURL

resp = yield treq.get(EncodedURL('https://httpbin.org/get?raw-value-here'))
```

Reply:
It very much should not be `EncodedURL`. IMO that API is dangerous because it doesn't handle percent-encoding for you. If you use it with untrusted input you'll get security injection vulnerabilities! Sometimes you need that lower-level interface, but it's not something I'd want to suggest. If you need it, you'll know.

Reply:
(I think this question speaks to the way that `EncodedURL` and `DecodedURL` are confusing names. Particularly since `URL is EncodedURL` and so seems like the default.)